.github/workflows/ci.yml

```diff
@@ -64,7 +64,71 @@ jobs:
           cd flink-autoscaler-plugin-jdbc
           mvn -B verify -Dit.skip=false
           cd ..
+  e2e_smoke_test:
+    name: HTTP Client smoke test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        http-client: [ "okhttp", "jdk", "jetty", "vertx" ]
+        version: ["v1_20"]
+        mode: ["native"]
+        namespace: ["default"]
+        java-version: ["21"]
```

Comment:
I think we should actually move the namespace , and java-version matrix also to the smoke_test and fix-it (remove-it) in the regular e2es.

The http-client, java-version, namespace matrix doesn't need all the different tests, it either works or it doesn't (famous last words).

Reply:
actually the namespace flink/default doesn't really need more than 1 test (regardless of other params...)

Reply:
> I think we should actually move the namespace , and java-version matrix also to the smoke_test and fix-it (remove-it) in the regular e2es.
> 
> 

I agree it's a bit strange. What I was intending to do was make the smallest change here and revisit it in a follow up hence the matrix of single values.

> actually the namespace flink/default doesn't really need more than 1 test (regardless of other params...)

Thanks that’s the kind of insight I need for restructuring🧩.

Reply:
@SamBarker , how would you like to proceed with this? As long as we are adding the matrix changes I suggested in a followup , this is good from my side.

Should we squash it and merge it or would you want to add the restructuring here as a "second" commit